### ISSUE TITLE
Negative length vectors when joining files in R

### ISSUE THREAD

**Issue by yaaminiv** [RobertsLab/resources#732]
I'm following @hputnam 's workflow to [annotate *C. virginica* coverage files](https://github.com/RobertsLab/resources/issues/727). In line 73 of [this script](https://github.com/hputnam/Geoduck_Meth/blob/master/RAnalysis/Scripts/methylation_analysis.R#L73), `full_join` is used to join sample coverage files and gene information, then `filter` is used to retain entries in which loci are assigned to the correct gene.

I tried using this in [my R Markdown script](https://github.com/fish546-2018/yaamini-virginica/blob/master/analyses/2019-08-14-Differentially-Methylated-Genes/2019-08-14-Differentially-Methylated-Gene-Analysis.Rmd) but encountered the following error:

```
temp <- full_join(sample1, geneAnnotationsCondensedUniprotGOterms, by="gene.chr") %>% filter(start >= gene.start & start <= gene.end)
Error in full_join_impl(x, y, by_x, by_y, aux_x, aux_y, na_matches, environment()) : 
  negative length vectors are not allowed
```

[Stack overflow](https://stackoverflow.com/questions/42479854/merge-error-negative-length-vectors-are-not-allowed) suggests this is happening because the resultant data frame has too many lines.

Any ideas on how to join these tables without encountering this error?

**Comment by kubu4** [comment]
Here are a few things to try:

1. Computer with more memory. I'm assuming you're still using the Mac Mini in 230? If yes, try running this on Ostrich (has 48GB of RAM, compared to the 16GB on the Mac Mini).

2. `filter` your data first and then perform the join. This should reduce the amount of data used in the join.

3. Split your input data frame into smaller data frames, join each of them independently and then concatenate the resulting join outputs into a single data frame.

**Comment by yaaminiv** [comment]
@kubu4 I am working on the Mac Mini. I tried splitting the input into smaller data frames (subsetted data from 1 chromosome), and got this error:

```
 temp <- full_join(sample1_NC_035780.1, geneAnnotationsCondensedUniprotGOterms, by="gene.chr") %>% filter(start >= gene.start & start <= gene.end)
Error: vector memory exhausted (limit reached?)
```

I'll try working on Ostrich.

**Comment by yaaminiv** [comment]
Update:

I tried `full_join` with the full data frames on Ostrich and ran into 2 issues:

1. The current version of `dplyr` is not compatible with the version of R on Ostrich (couldn't get `full_join` or `%>%` to work). Decided to side-step this by using `merge` with `all.x = TRUE` and `all.y = TRUE` and using `filter` in a separate step.
2. Negative length vector error

Was able to `merge` (seemingly) successfully when I annotated a single chromosome of 1 coverage file. Seeing how I have 10 samples with 10 chromosomes each (and the `merge` I started 2 hours ago isn't done...), I decided to switch over to `intersectBed` in R Markdown on genefish. That worked swimmingly.

TL;DR Don't know why I tried coming up with an R workaround when `bedtools` exists.

**Comment by hputnam** [comment]
@yaaminiv @kubu4 can you link to an example for the bedtools approach?
